src/Features/Core/Portable/EmbeddedLanguages/RegularExpressions/LanguageServices/RegexQuickInfoProvider.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.QuickInfo;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.RegularExpressions.LanguageServices
+{
+    [ExportEmbeddedLanguageQuickInfoProvider("Regex", new[] { LanguageNames.CSharp }, new[] { "Regex", "Regexp" }), Shared]
+    internal class RegexQuickInfoProvider : IEmbeddedLanguageQuickInfoProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public RegexQuickInfoProvider()
+        {
+        }
+
+        public QuickInfoItem? GetQuickInfo(QuickInfoContext context, SemanticModel semanticModel, SyntaxToken token)
+        {
+            //string url = "https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-language-quick-reference";
+            //return GetQuickInfo(context, semanticModel, token);
+            var item = QuickInfoItem.Create(token.Span, sections: new[]
+                {
+                    QuickInfoSection.Create(QuickInfoSectionKinds.Description, new[]
+                    {
+                        new TaggedText(TextTags.Punctuation, ""),
+                        new TaggedText(TextTags.Space, " "),
+                        new TaggedText(TextTags.Text,"Click"),
+                        new TaggedText(TextTags.Text, " Here", TaggedTextStyle.None, "https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-language-quick-reference", "https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-language-quick-reference"),
+                        new TaggedText(TextTags.Text, " to access the Regex Documentation page.")
```

Comment:
QuickInfo text needs to be localized. [This](https://github.com/dotnet/roslyn/blob/main/src/Features/Core/Portable/FeaturesResources.resx) file contains all localized strings for MS.CodeAnalysis.Feature project (where this source file belongs.)

You need to add a new data entry in the file, e.g.

```xml
  <data name="This_text_is_localized" xml:space="preserve">
    <value>This test is localized</value>
  </data>
```

Then it could be referenced as `FeaturesResources.This_text_is_localized` in your code, to replace those string literals